ext/com_dotnet/com_olechar.c

```diff
@@ -103,3 +103,62 @@ PHP_COM_DOTNET_API char *php_com_olestring_to_string(OLECHAR *olestring, size_t
 
 	return string;
 }
+
+BSTR php_com_string_to_bstr(zend_string *string, int codepage)
+{
+	BSTR bstr;
+	OLECHAR *olestring = NULL;
+	DWORD flags = codepage == CP_UTF8 ? 0 : MB_PRECOMPOSED | MB_ERR_INVALID_CHARS;
+	size_t mb_len = ZSTR_LEN(string);
+	int wc_len;
+
+	wc_len = MultiByteToWideChar(codepage, flags, ZSTR_VAL(string), (int)mb_len + 1, NULL, 0);
+	if (wc_len > 0) {
+		olestring = (OLECHAR*)safe_emalloc(wc_len, sizeof(OLECHAR), 0);
+		wc_len = MultiByteToWideChar(codepage, flags, ZSTR_VAL(string), (int)mb_len + 1, olestring, wc_len);
+	} else {
+		char *msg = php_win32_error_to_msg(GetLastError());
+
+		php_error_docref(NULL, E_WARNING,
+			"Could not convert string to unicode: `%s'", msg);
+		LocalFree(msg);
+
+		if (olestring == NULL) {
+			olestring = (OLECHAR*)emalloc(2 * sizeof(OLECHAR));
+		}
+		*olestring = *(olestring + 1) = '\0';
+	}
+
+	bstr = SysAllocStringLen(olestring, (UINT)(wc_len - 1));
+	efree(olestring);
```

Comment:
It seems rather weird that we first allocate a buffer of the right size, perform the wchar conversion into it, then allocate another buffer here, copy the converted string and then free it.

However, I did not spot any API that would just give us an uninitialized BSTR.

Reply:
Ah, it seems that `SysAllocStringLen(NULL, len)` would just do that.